src/app/qgsoptions.cpp

```diff
@@ -2259,6 +2259,7 @@ void QgsOptions::loadGdalDriverList()
   QStringList myDrivers;
   QStringList myGdalWriteDrivers;
   QMap<QString, QString> myDriversFlags, myDriversExt, myDriversLongName;
+  QMap<QString, bool> driversType; // true for raster, false for vector
```

Comment:
Would be slightly nicer if this were QMap< QString, QgsMapLayerType >, but that's not a blocker!

Reply:
Changed to use `QgsMapLayerType`.